src/executor/evaluate/mod.rs

```diff
@@ -601,6 +601,49 @@ async fn evaluate_function<'a, T: 'static + Debug>(
             }
         }
         .map(Evaluated::from),
+
+        Function::Substr { expr, start, count } => {
+            let string = match eval_to_str(expr).await? {
+                Nullable::Value(v) => v,
+                Nullable::Null => {
+                    return Ok(Evaluated::from(Value::Null));
+                }
+            };
+            let start = match eval_to_integer(start).await? {
+                Nullable::Value(v) => v,
+                Nullable::Null => return Ok(Evaluated::from(Value::Null)),
+            };
+
+            let count = match count {
+                Some(expr) => match eval_to_integer(expr).await? {
+                    Nullable::Value(v) => match v {
+                        x if x < 0 => return Err(EvaluateError::NegativeSubstrLenNotAllowed.into()),
+                        _ => v,
+                    },
+                    Nullable::Null => -1,
+                },
+                None => -1,
+            };
```

Comment:
Current `count` variable has two independent roles
1. whether `count` exists or not
2. number of chars to substr.

`ast` already properly provides as a form of `Option<Expr>` and you'll be able to convert it into `Option<usize>`.
Then, you can handle `Option<usize>` directly without merging two info into `i64` type.

Reply:
I don't know if it's okay to change type of `count` to` Option<usize>`,, :joy_cat: 
Because, to calculate the variable `e`, `count` may have to be added to the `start`, which causes type casting.
Is there any simpler way? XD